weedcoco/index/indexing.py

```diff
@@ -92,36 +101,39 @@ def _flatten(src, dst, prefix):
                     image["task_type"].add("bounding box")
             image["task_type"] = sorted(image["task_type"])
 
-        self.coco = coco
+        self.indexes = coco
 
     def generate_batches(self):
-
-        coco = self.coco
-        if "images" in coco and len(coco["images"]) > 0:
-            for i in range(ceil(len(coco["images"]) / self.index_batch)):
-                with open(f"{self.temp_index_dir}/index{str(i+1)}.json", "w") as f:
-                    for image in coco["images"][
-                        i
-                        * self.index_batch : min(
-                            len(coco["images"]), (i + 1) * self.index_batch
-                        )
-                    ]:
-                        json.dump({"index": {"_index": "weedid", "_type": "image"}}, f)
-                        f.write("\n")
-                        json.dump(image, f)
-                        f.write("\n")
-
-    def post_index(self):
-
-        files = [
-            file for file in os.listdir(self.temp_index_dir) if file.endswith(".json")
-        ]
-        for item in files:
-            requests.post(
-                f"{self.es_url}_bulk",
-                data=open(f"{self.temp_index_dir}/{item}", "rb").read(),
+        """
+        Split indexes into batches to reduce payload size of each request to ElasticSearch
+        """
+        indexes = self.indexes
+        if "images" in indexes and len(indexes["images"]) > 0:
+            images = indexes["images"]
+            for start in range(0, len(images), self.batch_size):
+                blobs = []
+                batch = images[start : start + self.batch_size]
+                for image in batch:
+                    blobs.append(
+                        '{"index": {"_index": "'
+                        + self.index_name
+                        + '", "_type": "image"}}'
+                    )
+                    blobs.append(json.dumps(image))
+                blobs.append("")
+                yield "\n".join(blobs)
+
+    def post_to_index(self):
+        """
+        Send post request to ElasticSearch
+        """
+        for index_batch in self.generate_batches():
+            res = requests.post(
+                urllib.parse.urljoin(self.es_url, "_bulk"),
+                data=index_batch,
                 headers={"content-type": "application/json"},
             )
+            print(res.headers, res.status_code, res.content)
```

Comment:
Will this raise an error if one/all of the indexing commands fail?

Is there a reason not to use the Python elasticsearch library to do this, as well as the batching, for us?